### PR TITLE
Ifpack2 BlockTriDi: updating example

### PR DESCRIPTION
**Pull request by lucbv**
Adding more logic to generate 3D matrices and have the block size as command line parameter. The driver still fails in some cases for unknown reasons at this time. One big issue at the moment is that only the y direction gets distributed across MPI ranks...

@trilinos/ifpack2 

## Motivation
This work allows us to reproduce more closely the use case for the BlockTriDiagSolver

## Testing
Tested on Frontier

### CODE REVIEW

**Review by jhux2 — CHANGES_REQUESTED** [review]
Needs to merge to `develop`, not `master`.

**Comment by lucbv** [comment]
Thanks @jhux2, I updated the base branch

**Comment by lucbv** [comment]
Ideally a github action should be able to detect that and block the PR automatically...

**Review by jhux2 — APPROVED** [review]

**Comment by jhux2** [comment]
@lucbv I was trying to dismiss my review and accidentally approved.  @kliegeois should review.

File: packages/ifpack2/example/BlockTriDi.cpp
Comment (jhux2) [inline packages/ifpack2/example/BlockTriDi.cpp:413]: Please put a comment as to what an "x_fiber" is.

File: packages/ifpack2/example/BlockTriDi.cpp
Comment (jhux2) [inline packages/ifpack2/example/BlockTriDi.cpp:172]: Since this an example, it would be good to have comment to explain this formula.

**Review by jhux2 — COMMENTED** [review]
Couple comments.

File: packages/ifpack2/example/BlockTriDi.cpp
Comment (kliegeois) [reply packages/ifpack2/example/BlockTriDi.cpp:413]: Does something like this is clear enough?

```
    // We compute the number of lines oriented along the x direction of the mesh.
    // This number is called line_per_x_fiber where a fiber refers to an initial
    // x line in the mesh before dividing it in sublines.
```

File: packages/ifpack2/example/BlockTriDi.cpp
Comment (kliegeois) [reply packages/ifpack2/example/BlockTriDi.cpp:172]: Does a comment like this make sense to you?

```
    // We enforce that the block matrix is SPD by setting the diagonal value
    // to a value which is greater than the sum of the absolute value of all the
    // entries on the corresponding row knowing that the off diagonal blocks are
    // set to minus identity.
```

File: packages/ifpack2/example/BlockTriDi.cpp
Comment (jhux2) [reply packages/ifpack2/example/BlockTriDi.cpp:413]: @kliegeois Yes, that explanation is clear, thank you.

File: packages/ifpack2/example/BlockTriDi.cpp
Comment (jhux2) [reply packages/ifpack2/example/BlockTriDi.cpp:172]: @kliegeois Yes, this is also clear.

**Comment by kliegeois** [comment]
Thanks for the review @jhux2 !

**Review by jhux2 — APPROVED** [review]
lgtm

**Comment by kliegeois** [comment]
@jhux2 what was the error?

**Pull request merged**
